tetris/index.js

```diff
@@ -0,0 +1,121 @@
+var express = require('express');
+var app = express();
+var http = require('http').Server(app);
+var io = require('socket.io')(http);
+var path = require('path');
+var util = require('util');
+
+var users = require(__dirname + '/js/userContainer.js')();
+var players = require(__dirname + '/js/playerContainer.js')();
+
+app.use("/scripts", express.static(__dirname + '/public/js'));
+app.use("/styles", express.static(__dirname + '/public/css'));
+
+app.get('/', function(req, res) {
+    res.sendFile('index.html', { root: __dirname });
+});
+
+io.on('connection', function(socket) {
+	
+	socket.on('disconnect', function() {
+		updatePlayers();
+		updateClientUserList();
+		
+		if (users.allUsers[socket.id].showDisconnectMessage)
+			io.emit('message', users.nameById(socket.id) + ' Has Disconnected');	
+	});
+	
+	socket.on('authenticate', function(token) {
+		var rejoining = false;
+		var userExists = users.isExistingUser(token);
+		
+		if (userExists && token !== socket.id) {
+			var clients = getClientIdList();
+			
+			for (var i = 0; i < clients.length; i++) {
+				if (clients[i] === token) {
+					io.to(token).emit('disconnect client', 'New connection made, disconnecting.');
+					io.emit('message', users.nameById(token) + " Has Reconnected as " + users.nameById(socket.id));
+					users.allUsers[token].showDisconnectMessage = false;
+					rejoining = true;
+				}
+			}
+		}
+		if (rejoining) {
+			// tie new user object to old player
+			// skip title (and maybe also lobby)
+		}
```

Comment:
Remove any code that you aren't currently using.
https://www.youtube.com/watch?v=JOAq3YN45YE